src/Bicep.Core/Analyzers/Linter/Rules/OutputsShouldNotContainSecretsRule.cs

```diff
@@ -1,14 +1,16 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
 using Bicep.Core.Diagnostics;
+using Bicep.Core.Navigation;
 using Bicep.Core.Semantics;
 using Bicep.Core.Semantics.Namespaces;
 using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem;
 using Microsoft.WindowsAzure.ResourceStack.Common.Extensions;
-using System;
-using System.Collections.Generic;
-using System.Linq;
 
 namespace Bicep.Core.Analyzers.Linter.Rules
```

Comment:
The rule currently checks for @secret but also does a heuristic check for the name "secret".  I would have expected the following to trigger the name heuristic:
```bicep
param secretParameter {  
  secretProperty: {    
    secretProperty: string
  }
}

output flagged object = secretParameter // no warning emitted
output notFlagged object = secretParameter // no warning emitted
output alsoNotFlagged object = secretParameter.secretProperty // no warning emitted
```
It could be argued that's a bad idea.  Feel free to tell me if you feel that way.

Reply:
I think the best approach here would be to augment the `SecretsInParamsMustBeSecureRule` to cover this. It looks like there's already a diagnostic on `param secretParameter` asking that it made `@secure()`, and I could update that rule to do the same name check on type properties of parameters. That way, the rule that checks outputs can assume that the secure type validation flag will be set on any value coming from a parameter.

That approach should also cover cases where an output value contains a resource property that the RP marked as secure, but it would miss cases like this:

```bicep
param obj object

output notFlagged string = obj.secretProperty
```

Reply:
@jeskew Would you mind turning this into a bug?  I'm rather busy this week or I'd do it myself.  Thanks!

Reply:
Opened #11083 to track.